src/main/scala/com/ing/wbaa/airlock/proxy/provider/aws/SignatureHelpers.scala

```diff
@@ -183,13 +185,31 @@ trait SignatureHelpers extends LazyLogging {
 
     val requestParameters = extractRequestParameters(httpRequest, version)
 
-    if (!requestParameters.isEmpty) {
-      logger.debug(s"Setting additional params for request $requestParameters")
+    httpRequest.method.value match {
+      case "POST" if !requestParameters.isEmpty =>
+        logger.debug(s"Setting additional params (as body) for request $requestParameters")
+        val requestParamsCombined =
```

Comment:
Why do we need 'if'? - for one element mkString will return only the element so it would be ok:
```suggestion
        val requestParamsCombined =
          requestParameters.asScala.map { case (k, v) =>
            s"$k=${v.asScala.head}"
          }.mkString("&")
```